This pull request adds support for json output the to `gh pr checks` …

areese opened the pull request:
…command.

It also adds support for a json format of `*` which is translated to all known fields.
Support for '*' json format affects all commands with a --json flag.

Add support for the --json and --jq flags to `gh pr checks`
Add support for '*' to be passed as fields for the --json flag to allow all fields to be returned

Fixes: #4671 

<!--
  Thank you for contributing to GitHub CLI!
  To reference an open issue, please write this in your description: `Fixes #NUMBER`
-->

areese committed:
Fixes #4672:  Add --json flag to gh prs check

This pull request adds support for json output the to `gh pr checks` command.
It also adds support for a json format of `*` which is translated to all known fields.
Support for '*' json format affects all commands with a --json flag.

Add support for the --json and --jq flags to `gh pr checks`
Add support for '*' to be passed as fields for the --json flag to allow all fields to be returned

cliAutomation commented:
Hi! Thanks for the pull request. Please ensure that this change is linked to an issue by mentioning an issue number in the description of the pull request. If this pull request would close the issue, please put the word 'Fixes' before the issue number somewhere in the pull request body. If this is a tiny change like fixing a typo, feel free to ignore this message.

areese committed:
Fixes #4672:  Add --json flag to gh prs check

Cleanup the code for generating the json, separate json printing and table printing into 2 separate functions.
Create a summary structure that can be used for collecting the passing/failing/skipping/pending data.
Move summary generation into a function that operates on the summary structure.

Add passing/failing/skipping/pending to the top level json output

Valenteguardado commented:
> for
korawit1188:accessible-survey

vilmibm commented:
We're still figuring out what design we want for this and, given the age of this PR, feel it's best to close it for now until we have a proper spec for this work. Thanks for the PR 🙏
